### PR TITLE
feat(rpc): basic eth filter support

### PR DESCRIPTION
**Pull request by mattsse**
Add some basic eth_filter handling for creating and removing filters.
needs additional work for implementing the actual change logic.

There's a problem in the client proc-macro that fails with the SubscriptionId type due to lifetime.

I'll create a good first issue to roll a compatible FilterId type instead.

### CODE REVIEW

**Comment by gakonst** [comment]
this pr needs more work right?

**Comment by mattsse** [comment]
need to implement all poll updates but this PR includes the basics

both this and the pubsub PR are kinda blocked by this #1317 
I guess to unblock I'll move block subscribers to a trait first.

**Review by onbjerg — APPROVED** [review]

**Pull request merged**
